services/src/main/groovy/org/jfrog/artifactory/client/ArtifactoryClientBuilder.java

```diff
@@ -156,6 +164,10 @@ private CloseableHttpClient createClientBuilder(URI uri) {
         for (HttpRequestInterceptor httpRequestInterceptor : requestInterceptorList) {
             artifactoryHttpClient.addInterceptorLast(httpRequestInterceptor);
         }
+
+        artifactoryHttpClient.addRequestInterceptor(httpProcessor);
+        artifactoryHttpClient.addResponseInterceptor(httpProcessor);
```

Comment:
Can you please share some information about the above two lines?
What's the goal of adding the ```httpProcessor``` as request and response interceptors here?
You already exposed an API that adds the interceptors below, so why do you force setting them here inside the builder? 
![image](https://user-images.githubusercontent.com/7830056/161436828-6e2a258a-d417-4dd6-8844-dd8708859994.png)

Reply:
The above two lines are the ones that actually assign the processor to the internal ArtifactoryHttpClient object which is used to built the client, all of this happens internally on  createClientBuilder(URI uri) method with no access from outside, hence removing the two of them will resolve that the processor won't be assigned to the client upon creation (please take a look at the complete implementation of the method and how it is used for the client creation).
Thank you 😊 .